tests/Regressions.hs

```diff
@@ -71,6 +76,48 @@ propEqAfterDelete (Keys keys) =
 mapFromKeys :: [Int] -> HM.HashMap Int ()
 mapFromKeys keys = HM.fromList (zip keys (repeat ()))
 
+------------------------------------------------------------------------
+-- Issue #254
+
+data KC = KC Int
+  deriving (Eq, Ord, Show)
+instance Hashable KC where
+  hashWithSalt salt _ = salt
+
+issue254Lazy :: Assertion
+issue254Lazy = issue254LazyLambda 2
+
+-- Important that oldV is not hoisted out by optimisation, so use NOINLINE
+{-# NOINLINE issue254LazyLambda #-}
+issue254LazyLambda :: Int -> Assertion
+issue254LazyLambda i = do
+  _ <- return ()
```

Comment:
What does this line do?

Reply:
Puts the let binding under a lambda, so it doesn't get hoisted out and shared in the root. Super subtle. I updated the comments to say so.